### ISSUE TITLE
Feature: cor.test results

### ISSUE THREAD

**Issue by drsimonj** [tidymodels/corrr#44]
Results of of `cor.test()` are considered useful by many (e.g., p-values). However, at present, there's no foreseeable way to incorporate his information. One thought is to add this information as attributes to a cor_df. 

Either way, here's some example code that tidies the output of `cor.test()`, which could replace the use of `cor` in `correlate()` if used correctly:

```
library(dplyr)

d <- mtcars
var_pairs <- t(combn(names(d), 2)) %>%
  as_data_frame() %>% 
  setNames(c("x", "y"))

var_pairs %>% 
  dplyr::mutate(r.test = purrr::map2(x, y, ~ stats::cor.test(d[[.x]], d[[.y]])),
                r.test = purrr::map(r.test, broom::tidy)) %>%
  tidyr::unnest(r.test)
```

**Comment by BobMuenchen** [comment]
Nice! I'll have to study map and map2 more to understand how they're doing that, but I love the result!

**Comment by sebastiansauer** [comment]
That's helpful, Thanks!

**Comment by rubenarslan** [comment]
I once made a function to get the stretched format based on the `Hmisc::rcorr` results. Bit old, as you can see through the use of `reshape2::melt`

```r
corr_pipe = .  %>% as.matrix() %>% 
Hmisc::rcorr(.) %>% {
  left_join(left_join(.$r %>% reshape2::melt() %>% rename(r = value), 
                      .$n %>% reshape2::melt() %>% rename(n = value), by = c("Var1", "Var2")),
            .$P %>% reshape2::melt() %>% rename(p = value), 
            by = c("Var1", "Var2"))
  }
```

**Comment by jsta** [comment]
Building off the top comment, here is a way to get a p.value matrix with the same dimensions as `shave(correlate(d))`:

```r 
library(dplyr)

d <- mtcars[,1:3]

var_pairs <- t(combn(names(d), 2)) %>%
  as_data_frame() %>% 
  setNames(c("x", "y"))

p_values <- var_pairs %>% 
  dplyr::mutate(r.test = purrr::map2(x, y, ~ stats::cor.test(d[[.x]], d[[.y]])),
                r.test = purrr::map(r.test, broom::tidy)) %>%
  tidyr::unnest(r.test)

res <- matrix(NA, 
              nrow = length(names(d)), 
              ncol = length(names(d)))
res[lower.tri(res)] <- p_values$p.value
res <- data.frame(res, stringsAsFactors = FALSE, row.names = names(d))
(res <- setNames(res, names(d)))
```

**Comment by Tazinho** [comment]
Just had a similar issue at work and tried to write the function in a general form without dependencies. Maybe it's useful. Though, not sure if a general function wouldn't fit better into purrr or similar packages:

``` r

mix_all <- function(df, fun_xy, sym = TRUE, diagonal = NA, rowname = TRUE) {
  
  out <- data.frame()
  
  for (i in seq_along(df)) {
    for (j in seq_along(df)) {
      
      if(sym & i > j) {
        out[i, j] <- out[j,i]
        next
      }
      
      out[i, j] <- fun_xy(df[[i]], df[[j]])
    }
  }
  
  if(!is.null(diagonal)) {
    out[.row(dim(out)) == .col(dim(out))] <- diagonal
  } 
  
  names(out) <- names(df)
  
  if(rowname) {
    out[seq_along(out)+1] <- out 
    out[1] <- names(df)
    names(out)[1] <- "rowname"
  }
    
  out
}

mix_all(iris[1:4], function(x, y) cor.test(x, y)$p.value) 
#>        rowname  Sepal.Width Petal.Length  Petal.Width Petal.Width.1
#> 1 Sepal.Length           NA 1.518983e-01 1.038667e-47  2.325498e-37
#> 2  Sepal.Width 1.518983e-01           NA 4.513314e-08  4.073229e-06
#> 3 Petal.Length 1.038667e-47 4.513314e-08           NA  4.675004e-86
#> 4  Petal.Width 2.325498e-37 4.073229e-06 4.675004e-86            NA
```

<sup>Created on 2020-01-07 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by brshallo** [comment]
See [my post](https://www.bryanshalloway.com/2020/06/03/tidy-2-way-column-combinations/) containing similar solutions to above but with *lots of commentary* -- maybe useful for @BobMuenchen or ppl interested in walk-through/explanation of (possible) steps. (However code at post is less elegant than @drsimonj initial solution and is negligent with memory -- as creates duplicates of vectors of data for each comparison).

I am a fan of [94](https://github.com/tidymodels/corrr/issues/94#issue-577377489)  as well as @Tazinho note above and think a function (in `purrr`, `tidyr`, or elsewhere) or more official documentation on 'best-practice' for arbitrary pairwise operations in the tidyverse could be helpful.

**Comment by juliasilge** [comment]
Closed in #116 thanks to @jameslairdsmith:

``` r
library(corrr)

calc_cor_test <- function(vec_a, vec_b){
  cor.test(vec_a, vec_b)$p.value
}

colpair_map(mtcars, calc_cor_test)
#> # A tibble: 11 x 12
#>    rowname       mpg       cyl      disp       hp     drat        wt     qsec
#>    <chr>       <dbl>     <dbl>     <dbl>    <dbl>    <dbl>     <dbl>    <dbl>
#>  1 mpg     NA         6.11e-10  9.38e-10  1.79e-7  1.78e-5  1.29e-10  1.71e-2
#>  2 cyl      6.11e-10 NA         1.80e-12  3.48e-9  8.24e-6  1.22e- 7  3.66e-4
#>  3 disp     9.38e-10  1.80e-12 NA         7.14e-8  5.28e-6  1.22e-11  1.31e-2
#>  4 hp       1.79e- 7  3.48e- 9  7.14e- 8 NA        9.99e-3  4.15e- 5  5.77e-6
#>  5 drat     1.78e- 5  8.24e- 6  5.28e- 6  9.99e-3 NA        4.78e- 6  6.20e-1
#>  6 wt       1.29e-10  1.22e- 7  1.22e-11  4.15e-5  4.78e-6 NA         3.39e-1
#>  7 qsec     1.71e- 2  3.66e- 4  1.31e- 2  5.77e-6  6.20e-1  3.39e- 1 NA      
#>  8 vs       3.42e- 5  1.84e- 8  5.24e- 6  2.94e-6  1.17e-2  9.80e- 4  1.03e-6
#>  9 am       2.85e- 4  2.15e- 3  3.66e- 4  1.80e-1  4.73e-6  1.13e- 5  2.06e-1
#> 10 gear     5.40e- 3  4.17e- 3  9.64e- 4  4.93e-1  8.36e-6  4.59e- 4  2.43e-1
#> 11 carb     1.08e- 3  1.94e- 3  2.53e- 2  7.83e-7  6.21e-1  1.46e- 2  4.54e-5
#> # … with 4 more variables: vs <dbl>, am <dbl>, gear <dbl>, carb <dbl>
```

<sup>Created on 2020-11-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>
